bayesian_cnn_prometheus/analysis/similarity_comparer.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+
+from bayesian_cnn_prometheus.evaluation.utils import load_nifti_file # refactor idea: move this function to more 'general' directory
+from skimage import metrics as sk_metrics
+from typing import Dict
+
+
+class SimilarityComparer:
+    def __init__(self, lesion_mask_path: str, variance_mask_path: str):
+        """
+        Creates SimilarityComparer object to compare generated variance mask with lesion mask.
+        :param lesion_mask_path: path to a file with original lesion mask
+        :param variance_mask_path: path to a file with generated variance mask
+        """
+        self.lesion_mask_path = lesion_mask_path
+        self.variance_mask_path = variance_mask_path
+        self.lesion_mask = load_nifti_file(lesion_mask_path)
+        self.variance_mask = self.normalize(load_nifti_file(variance_mask_path))
+        self.metrics = {}
+
+    def perform_analysis(self, print_metrics: bool = False):
+        """
+        Assigns the metrics describing lesion and variance similarity.
+        :param print_metrics: if True, prints all of the numeric metrics
+        """
+
+        self._assign_dice_coefficient()
+        self._assign_hausdorff_distance()
+        self._assign_jaccard_index()
+
+        if print_metrics:
+            self._print_metrics()
+
+    def get_metrics(self) -> Dict:
+        """
+        Returns similarity metrics.
+        :return: a dictionary of metrics
+        """
+        if self.metrics:
+            return self.metrics
```

Comment:
Uwaga, czepiam się :D 
Jeśli pole metrics nie jest "prywatne" to ten get nic nie zmienia, więc mogłoby go nie być. W sensie wywołanie similarity_comparer.metrics jest równoznacznie z wywołaniem similarity_comparer.get_metrics(), nie ma żadnego info że nic  tam nie ma jak jest None, ani nic.